core/variant/variant_op.h

```diff
@@ -923,7 +923,10 @@ class OperatorEvaluatorStringFormat<S, void> {
 	static inline void validated_evaluate(const Variant *left, const Variant *right, Variant *r_ret) {
 		bool valid = true;
 		String result = do_mod(*VariantGetInternalPtr<S>::get_ptr(left), &valid);
-		ERR_FAIL_COND_MSG(!valid, result);
+		if (unlikely(!valid)) {
+			*VariantGetInternalPtr<String>::get_ptr(r_ret) = *VariantGetInternalPtr<S>::get_ptr(left);
+			ERR_FAIL_MSG(result);
```

Comment:
This is a pre-existing issue, but the error message formatting won't match the rule I stated in my other review comment.

I think we might want to make it similar to what `vformat` does with the lowercase sub-sentence that `sprintf` returns:
```
        ERR_FAIL_COND_V_MSG(error, String(), String("Formatting error in string \"") + p_text + "\": " + fmt + ".");
```

E.g.:
```suggestion
			ERR_FAIL_MSG(vformat("String formatting error: %s.", result));
```

Reply:
Same thing for the other `ERR_FAIL_MSG(result);`.